packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1259,6 +1259,31 @@ async function renderToHTMLOrFlightImpl(
         `by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error`
     )
 
+    // Get the unique expressions so we can print them.
+    const expressions = Array.from(
+      new Set(
+        staticGenerationStore.prerenderState.dynamicAccesses.map(
+          ({ expression }) => expression
+        )
+      )
+    )
+      .sort()
+      .join(', ')
+    warn(`The following dynamic usage was detected: ${expressions}`)
+
+    // Get the unique stack traces and print them.
+    const stacks = Array.from(
+      new Set(
+        staticGenerationStore.prerenderState.dynamicAccesses
+          .map(({ stack }) => stack)
+          .filter((stack): stack is string => typeof stack === 'string')
+          .map((stack) => stack.replace(/^Error: /, ''))
+      )
+    ).sort()
+    for (const stack of stacks) {
+      error(stack)
+    }
```

Comment:
seems noisy for something that isn't an error case. Imagine you are happy with the level of dynamism of the app, you probably don't want logs on every build or revalidate that remind you of all the places you are triggering dynamic bailouts.

Maybe this is something we write to a manifest instead and there is some tooling you can do to print nicely dynamic usages after the build. And vercel can highlight this as part of the build output if you want to go investigate it after the fact.

Reply:
This isn't going to log every dynamic bailout, it will only log when there was a dynamic bailout but no postponed state generated, indicating that there was a errant `try/catch` present and React wasn't able to identify it as a postpone error.